src/pudl/analysis/service_territory.py

```diff
@@ -0,0 +1,605 @@
+            tables.
+
+    Returns:
+        geopandas.GeoDataFrame
+
+    """
+    ba_gdf = (
+        balancing_authority_counties(
+            ba_ids=ids,
+            st_eia861=st_eia861,
+            ba_assn_eia861=ba_assn_eia861,
+            limit_by_state=limit_by_state,
+        )
+        .pipe(
+            add_geometries,
+            census_gdf,
+            dissolve=dissolve,
+            dissolve_by=["report_date", "balancing_authority_id_eia"]
+        )
+    )
+    return ba_gdf
+
+
+def compile_geoms(pudl_out,
+                  census_counties,
+                  entity_type,  # "ba" or "util"
+                  dissolve=False,
+                  limit_by_state=True,
+                  save=True):
+    """
+    Compile all available utility or balancing authority geometries.
+
+    Args:
+        pudl_out (pudl.output.pudltabl.PudlTabl): A PUDL output object, which will
+            be used to extract and cache the EIA 861 tables.
+        census_counties (geopandas.GeoDataFrame): A GeoDataFrame containing the county
+            level US Census DP1 data and county geometries.
+        entity_type (str): The type of service territory geometry to compile. Must be
+            either "ba" (balancing authority) or "util" (utility).
+        dissolve (bool): Whether to dissolve the compiled geometries to the
+            utility/balancing authority level, or leave them as counties.
+        limit_by_state (bool): Whether to limit included counties to those with
+            observed EIA 861 data in association with the state and utility/balancing
+            authority.
+        save (bool): If True, save the compiled GeoDataFrame as a GeoParquet file before
+            returning. Especially useful in the case of dissolved geometries, as they
+            are computationally expensive.
+
+    Returns:
+        geopandas.GeoDataFrame
+
+    """
+    if (entity_type == "util") and limit_by_state:
+        logger.info("Limiting territory by state is not yet supported for utilities.")
+        return None
+    logger.info(
+        f"Compiling {entity_type} geometries with {dissolve=} and {limit_by_state=}.")
+    # Run the interim EIA 861 ETL and get some dataframes:
+    st_eia861 = pudl_out.service_territory_eia861()
+    ba_eia861 = pudl_out.balancing_authority_eia861()
+    ba_assn_eia861 = pudl_out.balancing_authority_assn_eia861()
+
+    ids = {
+        "ba": ba_eia861.balancing_authority_id_eia.unique(),
+        "util": st_eia861.utility_id_eia.unique(),
+    }
+    funcs = {
+        "ba": balancing_authority_geometries,
+        "util": utility_geometries
+    }
+
+    # Identify all Utility IDs with service territory information
+    geom = funcs[entity_type](
+        ids=ids[entity_type],
+        st_eia861=st_eia861,
+        ba_assn_eia861=ba_assn_eia861,
+        census_gdf=census_counties,
+        dissolve=dissolve,
+        limit_by_state=limit_by_state,
+    )
```

Comment:
As it is right now, a request for *any* of the 861 dataframes results in all of them being generated, since it's not a database pull, it's running the whole ETL, so I wanted to be able to use the same `pudl_out` object for both cases.